examples/step-12/step-12.cc

```diff
@@ -470,8 +471,17 @@ namespace Step12
   template <int dim>
   void AdvectionProblem<dim>::solve()
   {
-    SolverControl                    solver_control(1000, 1e-12);
-    SolverRichardson<Vector<double>> solver(solver_control);
+    SolverControl solver_control(1000, 1e-12);
```

Comment:
Out of curiosity, does the need for an increased basis length disappear if you replace `1e-12` by `1e-6 * right_hand_side.l2_norm()`? Using an absolute tolerance is almost never the right approach, and we've tried to move away from it over the years (but apparently not yet for this program). See also #9596.

Reply:
Thanks for the quick review!

I just tested it. The number of iterations is reduced if I use the relative error tolerance without changing the basis in comparison to the usage of the absolute tolerance __without__ changing the basis. But it it is still higher if compared to the case using absolute tolerance __and__ the larger basis. 

In numbers: Iterations in Cycle 4
1. Absolute tolerance and standard basis : 185
2. Relative tolerance and standard basis: 138 
3. Absolute tolerance and larger basis: 44
4. Relative tolerance and larger basis: 42

Moreover using a relative tolerance allows for more refinement cycles, i.e. the solver still converges. 
What do you think should we enlarge the basis and use the relative tolerance in agreement with the issue you linked?

Reply:
I think it would be good to use the relative tolerance and the larger basis. In practice, one cannot use overly large bases, and anything beyond 50-80 is probably affected by roundoff errors to the extent that optimality is still lost even with the present orthogonalization strategy (iterated modified Gram-Schmidt), so one has to accept some cost. I would suggest to fix the basis size to 100 at most.

Reply:
Yes, relative tolerance and <=100 basis size sounds good to me as well!

Reply:
@kronbichler @bangerth Thanks for the feedback I checked and 85 basis vectors yield the same number of iterations as 100, so I reduced the number.

I was wondering if it made sense to transfer the solution from the coarser grid to the finer grid in each cycle to get a good starting vector for the GMRES. I found the following post on the mailing list https://groups.google.com/g/dealii/c/DKigsRj6I5g/m/irafcOzNAAAJ 
It seems that for CG it does not make sense.

Reply:
Yes, it's not worth transfering the solution. 

As for 85 vs 100: If you don't hit the limit, it doesn't matter what you set the limit to. The number of iterations also wouldn't have changed if you had set the limit to 1000 :-) I think choosing 100 for the basis size is just fine. It's not worth trying to find the exact minimal number (which is equal to the number of iterations you need), since any small change in the benchmark will lead to a different determination anyway. So 100 is fine.

Reply:
Okay, thanks. I changed the basis size back to 100. @kronbichler do you like to add a comment about round-off errors in the doc? If not, from my side everything is fine and we can wait for the last commit to past the tests.

Reply:
Thank you @nils-schween - I think we should add some comments about round-off in the `SolverGMRES` class itself, as the current documentation does not mention orthogonalization at all, except for the tiny comment https://github.com/dealii/dealii/blob/64eb806c2a9c1251ac74e4eb30daad21b04f61a1/include/deal.II/lac/solver_gmres.h#L257-L260. I suggest we take this into account in a separate pull request. Let me open an issue for it so it gets documented.